### ISSUE TITLE
Global default gitignores are not respected

### ISSUE THREAD

**Issue by Julian** [ggreer/the_silver_searcher#594]
Git supports global gitignore files in two default locations: ~/.gitignore and $XDG_CONFIG_HOME/.git/config, and will look there even if core.excludesfile is unset (or even if it is).

**Comment by rmm5t** [comment]
This surprised me as well. As a quick fix, I just copied by `~/.gitignore` to `~/.agignore`

```
cp ~/.gitignore ~/.agignore
```

**Comment by rosetree** [comment]
Instead of copying you could just link `~/.agignore` to `~/.gitignore`, to prevent that multiple files need to be changed in the future.

```
ln -s ~/.gitignore ~/.agignore
```

**Comment by rmm5t** [comment]
@rosetree True, but I also wanted to add some extra ignores to `~/.agignore` that I don't want in `~/.gitignore`.  Overall, it would be nice if all possible gitignore files were respected by `ag`.

**Comment by jasonkarns** [comment]
I haven't looked at the source to see how ag generates the ignore list for git, but I'm assuming it actually reads the .gitignore file(s) itself. However, git exposes the ignored files via the CLI. So ag could get the git-ignore list using `git status --ignored --porcelain` which would take into account system-level ignores, user-level ignores, repo-level ignores, and directory-level ignores all at once.

**Comment by decaff** [comment]
Okay, Jason.  I implemented your very [clever idea](https://github.com/decaff/the_silver_searcher/commit/f46c0d544cb3638bc4de5c6910640c20cb1fccde).  Not tested on Unix.  But works for me on Windows.

**Comment by decaff** [comment]
And I found a problem with my straight-forward implementation of Jason's idea....

````git status --ignored --porcelain```` returns path data **relative** to the top-level directory of the repo.  If ag's current working directory is not the same as the top-level directory of the repo, a straight-forward implementation (which does not adjust for this difference in hierarchy), fails.
